### DISCUSSION TITLE
how to instal the "valhalla" route calculation engine under docker under virtualbox using ubuntu commands

### DISCUSSION THREAD

**Discussion by courageusemayssara** [gis-ops/docker-valhalla#48] [General]
Hello, At the moment I am installing the "Valhalla" route planner under docker using ubuntu commands, I have tried quite a few command lines available on a few models, but without success . My question is to give me the successful model if not the correct instructions to install Valhalla locally if possible. I would like to know how to make this router work locally on the computer?

for information I would like to give myself all the steps to follow to make Valhalla work. in my case since the platform which is installed on my computer is windows, so to avoid changing the system and all its procedures, I installed the "virtualbox" virtual machine, in which I installed docker and then I tried to locally install valhalla under docker under virtualbox using ubuntu commands. I tried quite a bit of code but it was not successful. and this is the reason why I asked my question on this forum.
Pending a favorable response please accept my best regards.

**Comment by nilsnolde** [comment]
can you elaborate what you tried exactly? I'm not at all sure about using docker inside a ubuntu virtualbox on windows, that might be troublesome. also since there's wsl2 linux subsystem available on all modern windows distros, virtualbox for ubuntu rarely makes sense anymore. so you could try that (wsl2).

or why not use docker on windows directly? usually needs quite some setup (changing some bios/uefi settings etc) but I've made it work before, so it's possible, also with our valhalla image here.

anyways, IMO those are things you should research, it's a little convoluted but totally doable with some research grease;)

**Comment by courageusemayssara** [comment]
thank you for answering me.

At first i tried directly to install valhalla under docker under windows but i had a lot of errors so i thought to install valhalla under ubuntu but i didn't want to modify my system it is the reason why i switched to virtual machine where i installed ubuntu operating system but similarly i had quite a few errors so i said to myself why not try installing valhalla under docker but this time using ubuntu commands since the first time I tried it under the windows operating system so it was at this point I got stuck.

Looking forward to a favorable response, please receive my best regards.

**Reply by nilsnolde** [reply]
now you'll have to spend the time to get it to work. that can take a few days, you'll just have to bite. IMO that's how you learn the most. so I suggest you go ahead and research to understand what exactly it _is_ you're trying. stackoverflow etc, but please not here, since this is not a project-specific discussion or issue.

**Comment by linneaanna** [comment]
I'm having the same problem as @courageusemayssara and is wondering if you got valhalla to run under docker with ubuntu??
